src/GitVersionCore/GitVersionCacheKeyFactory.cs

```diff
@@ -25,12 +25,29 @@ static string GetGitSystemHash(GitPreparer gitPreparer)
         {
             var dotGitDirectory = gitPreparer.GetDotGitDirectory();
 
+            // get repo .git directory -- not the worktree gitdir
+            dotGitDirectory = GetWorktreeParentDotGetDirectory(dotGitDirectory);
+
             // traverse the directory and get a list of files, use that for GetHash
             var contents = CalculateDirectoryContents(Path.Combine(dotGitDirectory, "refs"));
 
             return GetHash(contents.ToArray());
         }
 
+        static string GetWorktreeParentDotGetDirectory(string path)
+        {
+            if (string.IsNullOrEmpty(path)) return path;
+
+            var idx = path.LastIndexOf(@".git\worktrees\", StringComparison.Ordinal);
```

Comment:
Can't you use `Path.Combine()` instead, which uses the correct path separator according to the OS default?

Reply:
How about something like this?

```cs
        static string GetWorktreeParentDotGetDirectory(string path)
        {
            if (string.IsNullOrEmpty(path)) return path;

            var parentPath = Path.GetDirectoryName(path);
            if (string.IsNullOrEmpty(parentPath)) return path;

            var worktreesPath = Path.Combine(".git", "worktrees");

            if (!parentPath.EndsWith(worktreesPath, StringComparison.Ordinal))
                return path;

            return parentPath.Substring(0, parentPath.Length - 10);
        }
```

It is probably more correct too because it explicitly checks for the path to end in `.git/worktrees/whatever` instead of just containing `.git/worktrees/` somewhere in the path.

Reply:
Yep 👍